test/cluster/local_pvc_delete_test.go

```diff
@@ -0,0 +1,189 @@
+							Namespace: namespace,
+						},
+					}
+					Expect(DeleteCluster(k8sClient, ctx, aeroCluster)).ToNot(HaveOccurred())
+					Expect(CleanupPVC(k8sClient, aeroCluster.Namespace, aeroCluster.Name)).ToNot(HaveOccurred())
+				},
+			)
+
+			Context("When doing rolling restart", func() {
+				It("Should delete the local PVCs when deleteLocalStorageOnRestart is set", func() {
+					aeroCluster := createDummyAerospikeCluster(
+						clusterNamespacedName, 2,
+					)
+					aeroCluster.Spec.Storage.DeleteLocalStorageOnRestart = ptr.Bool(true)
+					aeroCluster.Spec.Storage.LocalStorageClasses = []string{storageClass}
+					Expect(DeployCluster(k8sClient, ctx, aeroCluster)).ToNot(HaveOccurred())
+
+					oldPvcInfoPerPod, err := extractClusterPVC(ctx, k8sClient, aeroCluster)
+					Expect(err).ToNot(HaveOccurred())
+
+					By("Updating pod metadata to trigger rolling restart")
+					aeroCluster.Spec.PodSpec.AerospikeObjectMeta = asdbv1.AerospikeObjectMeta{
+						Labels: map[string]string{
+							"test-label": "test-value",
+						},
+					}
+
+					Expect(updateCluster(k8sClient, ctx, aeroCluster)).ToNot(HaveOccurred())
+
+					By("Validating PVCs deletion")
+					validateClusterPVCDeletion(ctx, oldPvcInfoPerPod)
+				})
+
+				It("Should delete the local PVCs of only 1 rack when deleteLocalStorageOnRestart is set "+
+					"at the rack level", func() {
+					aeroCluster := createDummyAerospikeCluster(
+						clusterNamespacedName, 2,
+					)
+					rackConfig := asdbv1.RackConfig{
+						Racks: getDummyRackConf(1, 2),
+					}
+					storage := lib.DeepCopy(&aeroCluster.Spec.Storage).(*asdbv1.AerospikeStorageSpec)
+					rackConfig.Racks[0].InputStorage = storage
+					rackConfig.Racks[0].InputStorage.DeleteLocalStorageOnRestart = ptr.Bool(true)
+					rackConfig.Racks[0].InputStorage.LocalStorageClasses = []string{storageClass}
+					aeroCluster.Spec.RackConfig = rackConfig
+					Expect(DeployCluster(k8sClient, ctx, aeroCluster)).ToNot(HaveOccurred())
+
+					oldPvcInfoPerPod, err := extractClusterPVC(ctx, k8sClient, aeroCluster)
+					Expect(err).ToNot(HaveOccurred())
+
+					By("Updating pod metadata to trigger rolling restart")
+					aeroCluster.Spec.PodSpec.AerospikeObjectMeta = asdbv1.AerospikeObjectMeta{
+						Labels: map[string]string{
+							"test-label": "test-value",
+						},
+					}
+
+					Expect(updateCluster(k8sClient, ctx, aeroCluster)).ToNot(HaveOccurred())
+
+					By("Validating PVCs deletion")
+					validateClusterPVCDeletion(ctx, oldPvcInfoPerPod, clusterName+"-2-0")
+				})
+			})
+
+			Context("When doing upgrade", func() {
+				It("Should delete the local PVCs when deleteLocalStorageOnRestart is set", func() {
+					aeroCluster := createDummyAerospikeCluster(
+						clusterNamespacedName, 2,
+					)
+					aeroCluster.Spec.Storage.DeleteLocalStorageOnRestart = ptr.Bool(true)
+					aeroCluster.Spec.Storage.LocalStorageClasses = []string{storageClass}
+					Expect(DeployCluster(k8sClient, ctx, aeroCluster)).ToNot(HaveOccurred())
```

Comment:
Since we've already tested deploying the cluster with LocalStorageClasses and DeleteLocalStorageOnRestart configured from the start, in this test let's deploy the cluster without these settings, apply them afterwards, and verify that no rolling restart occurs. Once confirmed, we'll proceed with further testing.